src/main/scala/io/iohk/ethereum/network/discovery/Node.scala

```diff
@@ -0,0 +1,67 @@
+package io.iohk.ethereum.network.discovery
+
+import java.net._
+
+import akka.util.ByteString
+import org.spongycastle.util.encoders.Hex
+
+import scala.util.{Failure, Success, Try}
+
+trait Node {
+  val id: ByteString
+  val addr: InetSocketAddress
+
+  def toUri: URI = {
+    val host = addr.getHostName
+    val port = addr.getPort
+    new URI(s"enode://${Hex.toHexString(id.toArray[Byte])}@$host:$port")
+  }
+}
+
+case class NodeImpl(id: ByteString, addr: InetSocketAddress) extends Node
+
+object NodeImpl {
+  def fromUri(uri: URI): NodeImpl = {
+    val nodeId = ByteString(Hex.decode(uri.getUserInfo))
+    NodeImpl(nodeId, new InetSocketAddress(uri.getHost, uri.getPort))
```

Comment:
can this fail, I mean if URI is badly formatted
what happens if `getUserInfo` returns something that is not node Id?
should we use here similar logic as in `parseNode()`